content/en/blog/_posts/2024-06-21-consistent-read-from-cache.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: blog
+title: 'Kubernetes Consistent Reads from Cache: A Leap in Performance and Scalability'
+date: 2024-06-21
+slug: consistent-read-from-cache-beta
+author: >
+  Marek Siarkowicz (Google)
+---
+
+Kubernetes is renowned for its robust orchestration of containerized applications,
+but as clusters grow, the demands on the control plane can become a bottleneck.
+A key challenge has been ensuring consistent reads from the etcd datastore,
+often requiring resource-intensive quorum reads.
+
+Today, we're excited to announce a major improvement: **Consistent Reads from Cache**, graduating to Beta in Kubernetes 1.31.
+
+### Why Consistent Reads Matter
+
+Consistent reads are essential for ensuring that Kubernetes components,
+like the kubelet, have an accurate view of the cluster state.
+In the past, these reads involved a round-trip to etcd to guarantee consistency.
+While effective, this approach can lead to performance issues in large clusters,
+especially when kubelets need to frequently list pods scheduled on their nodes.
+
+### The Breakthrough: Caching with Confidence
+
+Kubernetes has long used a watch cache to optimize read operations.
+The watch cache stores a snapshot of the cluster state and receives updates through etcd watches.
+However, until now, it couldn't serve consistent reads directly, as there was no guarantee the cache was sufficiently up-to-date.
+
+**The Consistent Reads from Cache** feature addresses this by intelligently using
+etcd's "progress notifications." These notifications allow the watch cache to know
+precisely how current its data is.  When a consistent read is requested, the system
+first checks the progress notification. If the cache is sufficiently up-to-date,
+the read is served directly from the cache, bypassing the need for a full etcd round-trip.
+
+**Important Note:** To fully benefit from this feature, your Kubernetes cluster
+must be running etcd version 3.4.31+ or 3.5.13+.
+Older etcd versions lack the necessary functionality and are now considered deprecated for use with Kubernetes.
```

Comment:
Is this true? (Will the release announcement article and changelog include this deprecation; if it won't, then it's likely not to be an official / formal deprecation).

Reply:
True, but let's omit that